### ISSUE TITLE
nogis provider

### ISSUE THREAD

**Issue by jerabekjak** [storm-fsv-cvut/smoderp2d#84]
Provide a simple provider to prepare data without any GIS software.
This provider will only allow a simple geometry calculation on rectangular computing domain.

**Comment by jerabekjak** [comment]
initialize work in commit b3a864a5f7811d347007076fdd43b3d6d9a72134

**Comment by jerabekjak** [comment]
make in provider from scratch (without gis)
- [ ] mat_b
- [ ] mat_stream_reach
- [ ] mat_a
- [ ] mat_slope
- [ ] yllcorner
- [ ] mat_n
- [ ] NoDataValue
- [ ] cell_stream
- [ ] mat_dem
- [ ] array_points
- [ ] state_cell
- [ ] outletCells
- [ ] mat_inf_index
- [ ] mat_fd
- [ ] rr
- [ ] STREAM_RATIO
- [ ] bc
- [ ] combinatIndex
- [ ] poradi
- [ ] mfda
- [ ] vpix
- [ ] points
- [ ] mat_hcrit
- [ ] pixel_area
- [ ] rc
- [ ] streams
- [ ] maxdt
- [ ] mat_aa
- [ ] extraOut
- [ ] mat_reten
- [ ] mat_nan
- [ ] spix
- [ ] prtTimes
- [ ] br
- [ ] streams_loc
- [ ] mat_efect_cont
- [ ] itera
- [ ] outdir
- [ ] c
- [ ] temp
- [ ] sr
- [ ] mat_pi
- [ ] surface_retention
- [ ] type_of_computing
- [ ] mat_boundary
- [ ] r
- [ ] mat_ppl
- [ ] end_time
- [ ] xllcorner

**Comment by pesekon2** [comment]
I have somehow made it run till the end (more-or-less by using `dx` everywhere where `efect_cont` was used - based on [this comment](https://github.com/storm-fsv-cvut/smoderp2d/blob/issue_82_nogis_provider/smoderp2d/providers/nogis/__init__.py#L155)). See the output below. Do you think that this output makes sense? I did not change any other value from the unchecked ones.

```
2020-10-15 11:24:51,558 - SMODERP2D - INFO - -------------------------------------------------------------------------------- - [__init__:36]
2020-10-15 11:24:51,558 - SMODERP2D - INFO - Saving output data... - [runoff:386]
2020-10-15 11:24:51,560 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/cInfil_m3.asc saved - [__init__:64]
2020-10-15 11:24:51,560 - SMODERP2D - INFO - 	Array stats: min=1.75756698577 max=1.75756698577 mean=1.75756698577 - [__init__:67]
2020-10-15 11:24:51,561 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/cRain_m3.asc saved - [__init__:64]
2020-10-15 11:24:51,561 - SMODERP2D - INFO - 	Array stats: min=5.9 max=5.9 mean=5.9 - [__init__:67]
2020-10-15 11:24:51,561 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/mVel_m_s.asc saved - [__init__:64]
2020-10-15 11:24:51,561 - SMODERP2D - INFO - 	Array stats: min=0.0450552365292 max=0.0450552365292 mean=0.0450552365292 - [__init__:67]
2020-10-15 11:24:51,561 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/mrSearStr_Pa.asc saved - [__init__:64]
2020-10-15 11:24:51,561 - SMODERP2D - INFO - 	Array stats: min=0.435332305653 max=0.435332305653 mean=0.435332305653 - [__init__:67]
2020-10-15 11:24:51,562 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/mQsur_m3_s.asc saved - [__init__:64]
2020-10-15 11:24:51,562 - SMODERP2D - INFO - 	Array stats: min=5.14730126287 max=5.14730126287 mean=5.14730126287 - [__init__:67]
2020-10-15 11:24:51,562 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/cVsur_m3.asc saved - [__init__:64]
2020-10-15 11:24:51,562 - SMODERP2D - INFO - 	Array stats: min=4277.81388441 max=4277.81388441 mean=4277.81388441 - [__init__:67]
2020-10-15 11:24:51,562 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/mWLevel_m.asc saved - [__init__:64]
2020-10-15 11:24:51,562 - SMODERP2D - INFO - 	Array stats: min=0.0443899567302 max=0.0443899567302 mean=0.0443899567302 - [__init__:67]
2020-10-15 11:24:51,563 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/mQsheet_m3_s.asc saved - [__init__:64]
2020-10-15 11:24:51,563 - SMODERP2D - INFO - 	Array stats: min=0.002 max=0.002 mean=0.002 - [__init__:67]
2020-10-15 11:24:51,563 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/cSheetVOutM3.asc saved - [__init__:64]
2020-10-15 11:24:51,563 - SMODERP2D - INFO - 	Array stats: min=928.298739926 max=928.298739926 mean=928.298739926 - [__init__:67]
2020-10-15 11:24:51,563 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/mWLevelRill_m.asc saved - [__init__:64]
2020-10-15 11:24:51,563 - SMODERP2D - INFO - 	Array stats: min=0.0433899567302 max=0.0433899567302 mean=0.0433899567302 - [__init__:67]
2020-10-15 11:24:51,563 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/mQRill_m3_s.asc saved - [__init__:64]
2020-10-15 11:24:51,564 - SMODERP2D - INFO - 	Array stats: min=5.14530126287 max=5.14530126287 mean=5.14530126287 - [__init__:67]
2020-10-15 11:24:51,564 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/cRillVOut_m3.asc saved - [__init__:64]
2020-10-15 11:24:51,564 - SMODERP2D - INFO - 	Array stats: min=6699.03028896 max=6699.03028896 mean=6699.03028896 - [__init__:67]
2020-10-15 11:24:51,564 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/widthRill.asc saved - [__init__:64]
2020-10-15 11:24:51,564 - SMODERP2D - INFO - 	Array stats: min=0.787309675234 max=0.787309675234 mean=0.787309675234 - [__init__:67]
2020-10-15 11:24:51,564 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/cVIn_M3.asc saved - [__init__:64]
2020-10-15 11:24:51,564 - SMODERP2D - INFO - 	Array stats: min=4277.81388441 max=4277.81388441 mean=4277.81388441 - [__init__:67]
2020-10-15 11:24:51,565 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/surRet_M.asc saved - [__init__:64]
2020-10-15 11:24:51,565 - SMODERP2D - INFO - 	Array stats: min=0.5 max=0.5 mean=0.5 - [__init__:67]
2020-10-15 11:24:51,565 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/control/CumVRestL3.asc saved - [__init__:64]
2020-10-15 11:24:51,565 - SMODERP2D - INFO - 	Array stats: min=0.0 max=0.0 mean=0.0 - [__init__:67]
2020-10-15 11:24:51,565 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/massBalance.asc saved - [__init__:64]
2020-10-15 11:24:51,565 - SMODERP2D - INFO - 	Array stats: min=-5.47117906535e-13 max=-5.47117906535e-13 mean=-5.47117906535e-13 - [__init__:67]
2020-10-15 11:24:51,566 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/volRest_m3.asc saved - [__init__:64]
2020-10-15 11:24:51,566 - SMODERP2D - INFO - 	Array stats: min=3.64243301423 max=3.64243301423 mean=3.64243301423 - [__init__:67]
2020-10-15 11:24:51,566 - SMODERP2D - INFO - Raster ASCII output file tests/data/output/core/reachFid.asc saved - [__init__:64]
2020-10-15 11:24:51,566 - SMODERP2D - INFO - 	Array stats: min=2 max=2 mean=2.0 - [__init__:67]
2020-10-15 11:24:51,566 - SMODERP2D - INFO - -------------------------------------------------------------------------------- - [runoff:394]
2020-10-15 11:24:51,566 - SMODERP2D - INFO - Total computing time: 2.8393099308 - [runoff:396]

```

**Comment by pesekon2** [comment]
Closing the issue as it was solved in 8aaa3abb639f67aee25d3beedfdc34381c312e68 and f7bd397a5509b3806380d1b0560216d4932eafe4.
